jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/elements/complex/TimePicker.java

```diff
@@ -1,445 +1,461 @@
-    private static final String ACTIONS = ".v-picker__actions";
-
-    private DateTimeFormatter formatterTwelveHoursNoSeconds = DateTimeFormatter.ofPattern("K:mm a");
-    private DateTimeFormatter formatterTwelveHoursWithSeconds = DateTimeFormatter.ofPattern("K:mm:ss a");
-    private DateTimeFormatter formatterTwentyFourHoursNoSeconds = DateTimeFormatter.ofPattern("K:mm");
-    private DateTimeFormatter formatterTwentyFourHoursWithSeconds = DateTimeFormatter.ofPattern("K:mm:ss");
-    private DateTimeFormatter formatterResultDate = DateTimeFormatter.ofPattern("HH:mm");
-
-    @Override
-    public void setup(Field field) {
-        if (!fieldHasAnnotation(field, JTimePicker.class, TimePicker.class)) {
-            return;
-        }
-        JTimePicker j = field.getAnnotation(JTimePicker.class);
-        setup(j.root(), j.expandedRoot());
-    }
+public class TimePicker extends UIBaseElement<TimePickerAssert>
+    implements HasInit, HasColor, HasTheme, HasElevation, IsReadOnly, HasMeasurement {
 
-    public TimePicker setup(String rootLocator, String expandedRootLocator) {
-        if (isNotBlank(rootLocator)) {
-            root = rootLocator;
-        }
-        if (isNotBlank(expandedRootLocator)) {
-            expandedRoot = expandedRootLocator;
-        }
-        return this;
-    }
+    private static final String TITLE = ".v-picker__title";
+    private static final String TITLE_TIME = ".v-time-picker-title__time > div";
+    private static final String TITLE_BUTTONS_ALL = ".v-picker__title__btn";
+    private static final String TITLE_BUTTON_ACTIVE_CLASS = "v-picker__title__btn--active";
+    private static final String AM_PM_SWITCHERS = "[class*='__ampm'] .v-picker__title__btn";
+    private static final String CLOCK = ".v-time-picker-clock__inner";
+    private static final String CLOCK_NUMBERS = ".v-time-picker-clock__item";
+    private static final String CLOCK_NUMBERS_ACTIVE = ".v-time-picker-clock__item[class*='active']";
+    private static final String CLOCK_NUMBERS_DISABLED = ".v-time-picker-clock__item--disabled";
+    private static final String CLOCK_NUMBERS_ENABLED = ".v-time-picker-clock__item:not([class*='--disabled'])";
+    private static final String CLOCK_HAND = ".v-time-picker-clock__hand";
+    private static final String CLOCK_NUMBER_XPATH_TEMPLATE = "//span[contains(@class, 'v-time-picker-clock__item')]//span[text()='%s']";
 
-    public UIElement root() {
-        return $(root);
-    }
 
-    public UIElement expandedRoot() {
-        return $(expandedRoot);
-    }
+    public UIElement title;
+    public UIElement clock;
 
-    private UIElement expander() {
-        return root().find(EXPANDER);
-    }
-
-    private UIElement getCoreElement() {
-        UIElement coreElement = root();
-
-        if (expander().isExist()) {
-            coreElement = expandedRoot();
+    @Override
+    public void init() {
+        super.init();
+        title = core().find(TITLE);
```

Comment:
I need to discuss this, as I am not able to get StaleElementRefrenceException from JDI's UIElement in case an element defined by the locator is present on a page no matter if it was recreated or not.

Reply:
Fields are replaced with methods returning UIElements.